### ISSUE TITLE
Open-Api Custom Error

### ISSUE THREAD

**Issue by joeylgutierrez** [cdimascio/express-openapi-validator#561]
**Is your feature request related to a problem? Please describe.**
We are moving the company to use the lib but one issue we struggle with is identifying the validator errors to be able to convert them properly to our own custom errors.  The code to do so is kind of long and ugly just to know that it's an open-api error.

**Describe the solution you'd like**
I would love a custom error that has a boolean or function that returns a boolean to easy check the error if it's an open-api error.  Many libraries recently has been doing something like error.isAxiosError or hapi.isBoom().  maybe we can has error.isOpenApi?

**Describe alternatives you've considered**

**Additional context**

**Comment by cdimascio** [comment]
I'm open to suggestions here. Note that most of the errors returned from the generator are sourced from [here](https://github.com/cdimascio/express-openapi-validator/blob/master/src/framework/types.ts#L569-L746). Feel free to take a stab at it. The referenced code should get a good jumpstart, but there are some other error cases in the code that we'll need to sniffed out and updated

**Comment by setaman** [comment]
i see no improvement in the proposed solution for the use case i described in my issue. I still have to do the checks that i want to avoid. I could also write a function `isValidationError(error)` that checks, if the error is a validation error, so I have not much benefits from `error.isOpenApi`.

As an end user, I prefer my solution as it is simply more explicit. Furthermore, we already have the `onError` prop for `responseValidation`, which would also keep the interface more consistent.

**Comment by crudo** [comment]
@setaman You have to somehow identify the error. How do you want to do it without asking/checking?
Unique error property is fine in my eyes.

**Comment by setaman** [comment]
@crudo no need to check anything, if you have already defined an error handler that formats the validation error as you need. You can just propagate it in your error handler middleware. Otherwise you could write a functions (like i did to just improve the code) that identifies the error: 
````js
const isOpenAPISpecValidationError = (err) => err.errors && err.status,
````
So to call `isOpenAPISpecValidationError(err)` or `err.isOpenApi` makes not much difference. You still have to check.

**Comment by joeylgutierrez** [comment]
@crudo i believe he's attempting to just provide a custom error handler function so that he knows that if that function is called it can only be an openApi error so he wouldn't need to check.  In my case I just have a global error handler that catches several different error types from different libs and then builds custom errors from them.  I just need to know which is which

**Comment by crudo** [comment]
@joeylgutierrez Yep. I agree. My use case is closer to yours. I have multiple possible sources of errors and has to distinguish between them.

**Comment by cdimascio** [comment]
@setaman based on the discussion, it appears these are indeed two separate requests. I will reopen the other.

**Comment by raniantheone** [comment]
@joeylgutierrez I have similar requirement to yours. My current workaround is adding an error handling middleware right after openapi validator (since express middleware runs in order), but not sure if your project allows doing things this way.
